netshare/learn/setup_cross_chunks_data.py

```diff
@@ -0,0 +1,271 @@
+            raise ValueError(
+                '"encoding=cateogrical" can be only used for "type=(string | integer)"'
+            )
+        if "regex" in field:
+            return RegexField(
+                regex=field["regex"],
+                choices=[],  # Note: it will be filled *per chunk* in the per-chunk preprocessing
+                name=field_name,
+            )
+        return DiscreteField(
+            choices=list(pd.unique(this_df)),
+            name=field_name,
+        )
+
+    # Continuous Field: (float)
+    elif field["type"] == "float":
+        return ContinuousField(
+            name=field_name,
+            log1p_norm=field.get("log1p_norm", False),
+            norm_option=Normalization.from_config(field["normalization"]),
+            min_x=min(this_df) - EPS,
+            max_x=max(this_df) + EPS,
+            dim_x=1,
+        )
+
+    raise ValueError(
+        "Unable to build field from config (known field type / encoding): {}".format(
+            field
+        )
+    )
+
+
+def build_fields(
+    df: pd.DataFrame, word2vec_cols: Optional[List[str]]
+) -> Tuple[Dict[FieldKey, Field], Dict[FieldKey, Field]]:
+
+    session_key_fields: Dict[FieldKey, Field] = {
+        field_config_to_key(field): build_field_from_config(field, df, word2vec_cols)
+        for field in get_config(
+            ["pre_post_processor.config.metadata", "learn.session_key"]
+        )
+    }
+    timeseries_fields: Dict[FieldKey, Field] = {
+        field_config_to_key(field): build_field_from_config(field, df, word2vec_cols)
+        for field in get_config(
+            ["pre_post_processor.config.timeseries", "learn.timeseries"]
+        )
+    }
+
+    if get_config("global_config.n_chunks", default_value=1) > 1:
+        new_field = DiscreteField(name="startFromThisChunk", choices=[0.0, 1.0])
+        session_key_fields[key_from_field(new_field)] = new_field
+
+        for chunk_id in range(get_config("global_config.n_chunks")):
+            new_field = DiscreteField(
+                name="chunk_{}".format(chunk_id), choices=[0.0, 1.0]
+            )
+            session_key_fields[key_from_field(new_field)] = new_field
+
+    return session_key_fields, timeseries_fields
+
+
+def setup_cross_chunks_data(
+    big_df: pd.DataFrame, df_chunks: List[pd.DataFrame]
+) -> CrossChunksData:
+    """
+    This function splits the input data into chunks, and compute the .
+    """
+    embed_model, word2vec_cols = get_word2vec_model(big_df)
+    session_key_fields, timeseries_fields = build_fields(big_df, word2vec_cols)
+    flowkeys_chunkidx = get_flowkeys_chunkidx(df_chunks)
+    global_max_flow_len = get_global_max_flow_len(df_chunks)
+
+    return CrossChunksData(
+        embed_model=embed_model,
+        flowkeys_chunkidx=flowkeys_chunkidx,
+        global_max_flow_len=global_max_flow_len,
+        session_key_fields=session_key_fields,
+        timeseries_fields=timeseries_fields,
+    )
```

Comment:
Are we not storing `flowkeys_chunkdix` and other utility plots? These could be handy in terms of debugging/checking stats.

Reply:
Hmm that's a great idea.
I think that it makes sense to log it in `debug`, thus it will be shown only when we try to debug the app, but it will not be printed in standard executions